### PR TITLE
Update winrm and winrmcp dependencies

### LINKED ISSUE #3763: Crash during cancelled build using directory upload over WinRM
Crash after cancellation during directory upload over WinRM: https://gist.github.com/deltamualpha/06c8fd005fbe808bc55477caf9e650e1

packer.json file: https://gist.github.com/deltamualpha/01e9114aae9d3e0da4a5b73345c83ffe
```
$ packer -v
0.10.1
```

### PR DESCRIPTION
**Pull request by pkosel**
Updated winrm and winrmcp dependencies. Relevant unit and acceptance tests passed successfully.

`shell.Execute` didn't return a Command object.

See https://github.com/packer-community/winrmcp/blob/f1bcf36a69fa2945e65dd099eee11b560fbd3346/winrmcp/cp.go#L167 and https://github.com/masterzen/winrm/blob/54ea5d01478cfc2afccec1504bd0dfcd8c260cfa/winrm/shell.go#L10-L22 respectively.

Closes #3763

### CODE REVIEW

**Comment by veetow** [comment]
For what it's worth, this PR fixes an issue I was hitting with a very long-lived script provisioner on Windows. 👍

**Comment by kenmacleod** [comment]
Note, this change masterzen/winrm@d0501612972f7c061423081688281e53713f0bb9 introduced a header-response timeout.

This fixes an issue where WinRM was hanging during polling while "Waiting for WinRM to become available..." but the timeout appears to also occur during normal long operations.  In my case uploading VMWare Tools fails with this timeout.

I think the correct fix is pass Packer's `winrm_timeout` on requests that occur after the "Waiting for WinRM" loop.

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -348,11 +348,11 @@
...
 		{
-			"ImportPath": "github.com/masterzen/winrm/soap",
-			"Rev": "54ea5d01478cfc2afccec1504bd0dfcd8c260cfa"
+			"ImportPath": "github.com/masterzen/winrm",
+			"Rev": "ef3efbb97f99fc204bd9c7edf778a0dbd9781baf"
 		},
 		{
-			"ImportPath": "github.com/masterzen/winrm/winrm",
+			"ImportPath": "github.com/masterzen/winrm/soap",
 			"Rev": "54ea5d01478cfc2afccec1504bd0dfcd8c260cfa"
```
Comment (rickard-von-essen) [inline Godeps/Godeps.json]: Both `github.com/masterzen/winrm` and `github.com/masterzen/winrm/soap` should reference the same rev.

File: vendor/github.com/masterzen/winrm/client_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package winrm
```
Comment (rickard-von-essen) [inline vendor/github.com/masterzen/winrm/client_test.go]: Tests should not be included.

**Comment by rickard-von-essen** [comment]
Looks like the dependencies have manually been updated.

**Comment by rickard-von-essen** [comment]
I redid this change in #3947, it was a bit trickier than I initial thought. 

@pkosel @veetow it would be awesome if you could test that.

**Comment by rickard-von-essen** [comment]
Closing this in favour of #3947. Thanks for your effort!

**Pull request closed without merging**